causalpy/pymc_models.py

```diff
@@ -507,3 +507,295 @@ def fit(self, X, t, coords):
+            self.treatment_effect_type = [treatment_effect_type]
+        else:
+            self.treatment_effect_type = treatment_effect_type
+
+        # Defining the priors here
+        self.treatment_effect_param = (
+            {} if treatment_effect_param is None else treatment_effect_param
+        )
+
+        if "level" in self.treatment_effect_type:
+            if (
+                "level" not in self.treatment_effect_param
+                or len(self.treatment_effect_param["level"]) != 2
+            ):
+                self.treatment_effect_param["level"] = self.DEFAULT_LEVEL_PRIOR
+            else:
+                self.treatment_effect_param["level"] = self.treatment_effect_param[
+                    "level"
+                ]
+
+        if "trend" in self.treatment_effect_type:
+            if (
+                "trend" not in self.treatment_effect_param
+                or len(self.treatment_effect_param["trend"]) != 2
+            ):
+                self.treatment_effect_param["trend"] = self.DEFAULT_TREND_PRIOR
+            else:
+                self.treatment_effect_param["trend"] = self.treatment_effect_param[
+                    "trend"
+                ]
+
+        if "impulse" in self.treatment_effect_type:
+            if (
+                "impulse" not in self.treatment_effect_param
+                or len(self.treatment_effect_param["impulse"]) != 3
+            ):
+                self.treatment_effect_param["impulse"] = self.DEFAULT_IMPULSE_PRIOR
+            else:
+                self.treatment_effect_param["impulse"] = self.treatment_effect_param[
+                    "impulse"
+                ]
+
+    def build_model(self, X, y, coords):
+        """
+        Defines the PyMC model
+
+        :param X: An array of the covariates
+        :param y: An array of values representing our outcome y
+        :param coords: Dictionary of named coordinates for PyMC variables (e.g., {"obs_ind": range(n_obs), "coeffs": range(n_covariates)}).
+
+        Assumes the following attributes are already defined in self:
+            - self.timeline: the index of the column in X representing time.
+            - self.time_range: a tuple (lower_bound, upper_bound) for the intervention time.
+            - self.treatment_effect_type: a dictionary specifying which intervention effects to include and their priors.
+        """
+
+        with self:
+            self.add_coords(coords)
+
+            t = pm.Data("t", np.arange(len(X)), dims="obs_ind")
+            X = pm.Data("X", X, dims=["obs_ind", "coeffs"])
+            y = pm.Data("y", y, dims="obs_ind")
+
+            lower_bound = pm.Data("lower_bound", self.time_range[0])
+            upper_bound = pm.Data("upper_bound", self.time_range[1])
+
+            # --- Priors ---
+            treatment_time = pm.Uniform(
+                "treatment_time", lower=lower_bound, upper=upper_bound
+            )
+            beta = pm.Normal(
```

Comment:
Ah, ok,... so you are using a slope parameter just allow configuration in terms of mu, sigma... get you